frontend/src/layouts/BaseLayout.vue

```diff
@@ -278,13 +278,27 @@ export default defineComponent({
   setup() {
     const { advancedMode, isDark, language, supportedLanguages, toggleAdvancedMode, setLanguage, toggleDarkMode } =
       useSettingsStore();
-    const { avatar, isAccountSetup, isAccountSetupLegacy, isArgent, isLoading, network, userAddress, userDisplayName } =
-      useWalletStore();
+    const {
+      avatar,
+      isAccountSetup,
+      isAccountSetupLegacy,
+      isArgent,
+      isLoading,
+      network,
+      userAddress,
+      userDisplayName,
+      setLanguage: setWalletLanguage,
+    } = useWalletStore();
 
     const currentLanguage = ref({ label: language.value.label, value: language.value.value });
     const argentModalDismissed = ref(false);
     const showArgentModal = computed(() => isArgent.value && !argentModalDismissed.value);
     const version = window.logger.version;
+
+    const changeLanguage = (language: { label: string; value: string }) => {
```

Comment:
Is there a place to reuse types? This type exists in settings.ts

Reply:
`models.ts` is where we have type definitions